src/cirrus/lib/process_payload.py

```diff
@@ -208,26 +207,33 @@ def __call__(self, wfem) -> str | None:
             s3().upload_json(self, url)
 
             # create DynamoDB record
-            # -> this overwrites existing states other than PROCESSING
-            wfem.claim_processing(self["id"], payload_url=url)
+            # -> this overwrites existing states other than PROCESSING and CLAIMED
+            execution_name = wfem.claim_processing(
+                self["id"],
+                payload_url=url,
+                arn_base=arn,
+            )
 
             # invoke step function
             self.logger.debug("Running Step Function %s", arn)
-            exe_response = get_client("stepfunctions").start_execution(
+            get_client("stepfunctions").start_execution(
                 stateMachineArn=arn,
+                name=execution_name,
                 input=json.dumps(self.get_payload()),
             )
 
             # add execution to DynamoDB record
             wfem.started_processing(
                 self["id"],
-                exe_response["executionArn"],
+                f"{arn}:{execution_name}",
                 payload_url=url,
             )
 
             return self["id"]
         except ClientError as e:
             if e.response["Error"]["Code"] == "ConditionalCheckFailedException":
+                # TODO: may need to address this check for claim and processing
+                #       separately now that we are specifying the execution name
```

Comment:
You will, at least, need to handle a new error code for an already-existing execution with the given execution name. If you get that error you'll need to set the record to processing.

Reply:
@jkeifer -- My thought here is that this StepFunctions `ClientError` `Code="ExecutionAlreadyExists"` should be caught in an inner try-except, and then modify `StateDB.set_processing` to accept either `CLAIMED` or `PROCESSING` in the `ConditionExpression`.  There is some amount of race condition on getting the `set_processing` call if we don't make it idempotent.

Alternatively, we could add a specific `StateDB.already_processing` function which was used only in the case where we got the `ConditionalCheckFailedException` and we've already seen the `ExecAlreadyExists` error (last scenario in graph).

![retry-after-timeout-scenarios](https://github.com/user-attachments/assets/03b45377-ca4d-42e2-b780-7e4a0433e7a7)

I think making `set_processing` permissive to being either in `CLAIMED` or `PROCESSING` is the simplest solution, and will proceed with that.  Wanted to touch base here, as pushing race-conditions around the codebase feels worth a little extra discussion.

Reply:
My thinking on this is that in the race lost case we don't actually care about updating the state--we only want to set it to processing once, and we want that to be as early (as close to the step function start) as possible. I think implementing such behavior might necessitate blowing the exception handling apart to have nested try/except blocks for each expected exception.

Let me know if this comment doesn't give you enough direction and we can find a time to discuss further.

Reply:
That comment works.  It is in-line with what we discussed previously.  I think emitting the WorkflowEvent `ALREADY_PROCESSING` is probably good enough in the race-lost case.

Reply:
Alright.  `ALREADY_PROCESSING` is wired in, as well as `ALREADY_CLAIMED`.  The latter may be of interest as a symptom of other issues.